scripts/merge_users.py

```diff
@@ -0,0 +1,285 @@
+def merge_users(context, mongodb, username, password, authdb, host, port, loglevel, config, live):
+    """scout: manage interactions with a scout instance."""
+    coloredlogs.install(level=loglevel)
+
+    LOG.info("Running scout version %s", __version__)
+    LOG.debug("Debug logging enabled.")
+
+    mongo_config = {}
+    cli_config = {}
+    if config:
+        LOG.debug("Use config file %s", config)
+        with open(config, 'r') as in_handle:
+            cli_config = yaml.load(in_handle, Loader=yaml.FullLoader)
+
+    mongo_config['mongodb'] = (mongodb or cli_config.get('mongodb') or 'scout')
+
+    mongo_config['host'] = (host or cli_config.get('host') or 'localhost')
+    mongo_config['port'] = (port or cli_config.get('port') or 27017)
+    mongo_config['username'] = username or cli_config.get('username')
+    mongo_config['password'] = password or cli_config.get('password')
+    mongo_config['authdb'] = authdb or cli_config.get('authdb') or mongo_config['mongodb']
+    mongo_config['omim_api_key'] = cli_config.get('omim_api_key')
+
+
+    # always dryrun for now
+    dryrun = True
+    if live:
+        dryrun = False
+
+    LOG.info("Setting database name to %s", mongo_config['mongodb'])
+    LOG.debug("Setting host to %s", mongo_config['host'])
+    LOG.debug("Setting port to %s", mongo_config['port'])
+
+    valid_connection = check_connection(
+        host=mongo_config['host'],
+        port=mongo_config['port'],
+        username=mongo_config['username'],
+        password=mongo_config['password'],
+        authdb=mongo_config['authdb'],
+    )
+
+    LOG.info("Test if mongod is running")
+    if not valid_connection:
+        LOG.warning("Connection could not be established")
+        context.abort()
+
+    try:
+        client = get_connection(**mongo_config)
+    except ConnectionFailure:
+        context.abort()
+
+    database = client[mongo_config['mongodb']]
+
+    LOG.info("Setting up a mongo adapter")
+    mongo_config['client'] = client
+    adapter = MongoAdapter(database)
+
+    user_requests = []
+    event_requests = []
+    acmg_requests = []
+    clinvar_requests = []
+    clinvar_submission_requests = []
+
+    for oi_user_obj in adapter.user_collection.find({'_id': {'$type':'objectId'}}):
+        if not oi_user_obj.get('email'):
+            continue
+
+        LOG.info('===USER===')
+        oi_user_id = oi_user_obj.get('_id')
+        oi_user_email = oi_user_obj.get('email')
+        LOG.info("user: {}".format(oi_user_obj))
+
+        create_alt = False
+        alt_user_obj = adapter.user_collection.find_one({'_id': oi_user_email})
+        if not alt_user_obj:
+            create_alt = True
+            alt_user_obj = copy.deepcopy(oi_user_obj)
+            alt_user_obj['_id'] = oi_user_email
+        else:
+            LOG.info("alt user: {}".format(alt_user_obj))
+            merged_institutes = set()
+            merged_institutes.update(alt_user_obj.get('institutes') + oi_user_obj.get('institutes'))
```

Comment:
```suggestion
            merged_institutes.update(alt_user_obj.get('institutes',[]) + oi_user_obj.get('institutes', []))
```
This will fail if institutes is None or does not exist

Reply:
Same as below; I'll buy it since it is at least as good as before, and we can argue semantics of edge cases later. 😈